docs/source/pages/lightning.rst

```diff
@@ -45,11 +46,42 @@ The example below shows how to use a metric in your `LightningModule <https://py
             self.log('train_acc_epoch', self.accuracy)
 
 .. note::
+
     ``self.log`` in Lightning only supports logging of *scalar-tensors*. While the vast majority of metrics in torchmetrics returns a scalar tensor, some metrics such as
     :class:`~torchmetrics.ConfusionMatrix`, :class:`~torchmetrics.ROC`, :class:`~torchmetrics.MAP`, :class:`~torchmetrics.RougeScore` return outputs that are non-scalar
     tensors (often dicts or list of tensors) and should therefore be dealt with separately. For info about the return type and shape please look at the documentation for
     the ``compute`` method for each metric you want to log.
 
+.. note::
+
+    Modular metrics contain internal states that should belong to only one DataLoader. In case you are using multiple DataLoaders,
+    it is recommended to initialize a separate modular metric instances for each DataLoader and use them separately.
+
+    .. testcode:: python
+
+        class MyModule(LightningModule):
+
+            def __init__(self):
+                ...
+                self.val_acc_dl_0 = torchmetrics.Accuracy()
+                self.val_acc_dl_1 = torchmetrics.Accuracy()
+
+            def val_dataloader(self):
+                return [DataLoader(...), DataLoader(...)]
+
+            def validation_step(self, batch, batch_idx, dataloader_idx):
+                x, y = batch
+                preds = self(x)
+                ...
+
+                if dataloader_idx == 0:
+                    self.val_acc_dl_0(preds, y)
+                    self.log('val_acc', self.val_acc_dl_0)
+                elif dataloader_idx == 1:
+                    self.val_acc_dl_1(preds, y)
+                    self.log('val_acc', self.val_acc_dl_1)
```

Comment:
nit: this could be simplified if we recommend to initialize the metrics as a `nn.ModuleList`
```suggestion
                self.val_acc[dataloader_idx](preds, y)
                self.log("val_acc", self.val_acc[dataloader_idx])
```

Reply:
done!